dynamiqs/smesolve/smediffrax.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+import diffrax as dx
+import equinox as eqx
+import jax.numpy as jnp
+from jax import Array
+from jaxtyping import PyTree
+
+from ..core.abstract_solver import SMESolver
+from ..core.diffrax_solver import AdaptiveSolver, DiffraxSolver, EulerSolver
+from ..result import Saved, SMESaved
+from ..utils.utils.general import dag, trace, tracemm
+
+
+# state for the diffrax solver for SMEs
+class Y(eqx.Module):
+    rho: Array
+    dYt: Array
+
+
+class MeasurementTerm(dx.ControlTerm):
+    def prod(self, vf: dx.VF, control: dx.Control) -> dx.Y:
+        dW = control
```

Comment:
Why not just name it `dW` in the arguments? Would be more explicit IMO. This looks like there's additional logic.

Reply:
I like using the notations from diffrax API/source code in the overriden function signatures, I think it's neat to understand how our problem maps to the diffrax API. When I read
```python
def prod(self, vf: dx.VF, control: dx.Control) -> dx.Y:
```
I know the `prod` method of the diffrax `ControlTerm` is  reimplemented, while e.g.
```python
def prod(self, rho: dx.VF, dW: dx.Control) -> dx.Y:
```
is more obscure IMO.

In any case, when overriding a base class method it's best practice not to rename the arguments.